### PR TITLE
Update win_copy for #32677

### LINKED ISSUE #32677: win_copy does not support large files
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

win_copy

##### ANSIBLE VERSION

```
ccbm@amslabweb03.uniface.m4.local> ansible --version
ansible 2.4.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/h/CCbm/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### CONFIGURATION

DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles']

##### OS / ENVIRONMENT

N/A (Linux XXXXXXXXXXX.XXXXXXX.XX.XXXXX 3.10.0-327.18.2.el7.x86_64 #1 SMP Thu May 12 11:03:55 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
)

##### SUMMARY

win_copy by default does not use large file support when building a ZIP file that needs to be transferred. 
When doing a copy of a file larger than 4 Gb, this results in an error. 

##### STEPS TO REPRODUCE

```yaml
  - name: Copy Visual Studio installer to local machine
    win_copy:
      src: /cwnl/common/software/microsoft/VisualStudio_2015
      dest: C:\Temp\BuildMachineTemp\
```

##### EXPECTED RESULTS

File copy should work without problems
##### ACTUAL RESULTS

```
<!--- What actually happened? If possible run with extra verbosity (-vvvv) -->
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/win_copy.py", line 290, in _copy_zip_file
    zip_file = self._create_zip_tempfile(files, directories)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/win_copy.py", line 252, in _create_zip_tempfile
    zip_file.write(file_path, encoded_path, zipfile.ZIP_DEFLATED)
  File "/usr/lib64/python2.7/zipfile.py", line 1137, in write
    self._writecheck(zinfo)
  File "/usr/lib64/python2.7/zipfile.py", line 1102, in _writecheck
    raise LargeZipFile("Filesize would require ZIP64 extensions")
LargeZipFile: Filesize would require ZIP64 extensions
fatal: [amsbldw32i01.uniface.m4.local]: FAILED! => {
    "changed": true,
    "dest": "C:\\Temp\\BuildMachineTemp\\",
    "failed": true,
    "msg": "failed to create tmp zip file: Filesize would require ZIP64 extensions",
    "operation": "folder_copy",
    "src": "/cwnl/common/software/microsoft/VisualStudio_2015"
}
```

### PR DESCRIPTION
**Pull request by u625030**
SUMMARY

Fixes missing large file support in win_copy

Fixes: #32677 

ISSUE TYPE

Bugfix Pull Request

COMPONENT NAME

lib/ansible/plugins/action/win_copy.py

ANSIBLE VERSION

Stable-2.4

### CODE REVIEW

**Comment by jborean93** [comment]
Does this actually work for you, by default the compression and allowZip64 are set to `ZIP_STORED` and `True`. The only difference that this makes is that instead of just creating a zip file with no compression (`ZIP_STORED`), it compresses the archive (`ZIP_DEFLATED`).

I never set `ZIP_DEFLATED` as it requires zlib and I wanted to reduce the number of dependencies as much as possible. I can't find any documentation to say otherwise but I would have thought zipfile would be able to use the Zip64 extension regardless of the compression chosen.

I'll do some testing the background and see if t3hat is the case.

Edit 1: Looks like since Python 3.4 `allowZip64` is set to `true` but on 2.6 and 2.7 it is set to `false` potentially causing this issue.

File: lib/ansible/plugins/action/win_copy.py [hunk 1]
```diff
@@ -233,7 +233,7 @@ def _create_content_tempfile(self, content):
     def _create_zip_tempfile(self, files, directories):
         tmpdir = tempfile.mkdtemp()
         zip_file_path = os.path.join(tmpdir, "win_copy.zip")
-        zip_file = zipfile.ZipFile(zip_file_path, "w")
+        zip_file = zipfile.ZipFile(zip_file_path, "w", zipfile.ZIP_DEFLATED, True)
```
Comment (jborean93) [inline lib/ansible/plugins/action/win_copy.py]: Can you change this to just set `allowZip64=True` and not set the compression.

**Review by jborean93 — CHANGES_REQUESTED** [review]
Have tested this with Python 2.6, 2.7 and 3.6 and it works properly. Once you've removed the compression setting I'm happy to merge it in.

File: lib/ansible/plugins/action/win_copy.py (hunk 1 above)
Comment (u625030) [reply lib/ansible/plugins/action/win_copy.py]: Let me run a test here too; once that passes with your suggested setting, I will do a new commit. Stay tuned.

**Comment by u625030** [comment]
I ran some tests here with the ZIP_STORE setting rather than ZIP_DEFLATE and that works ok. Commited the ZIP_STORE setting.

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
Errors are unrelated to the code changes, thanks for the fix @u625030.

**Comment by jborean93** [comment]
Fix cherry-picked to stable-2.4 https://github.com/ansible/ansible/commit/612d9e04d73eaa777689bf5df927502de4ec3f09, will be in the 2.4.2rc1 release.

**Pull request merged**